subsys/net/ip/tcp.c

```diff
@@ -1406,6 +1406,15 @@ int net_tcp_parse_opts(struct net_pkt *pkt, int opt_totlen,
 		opt_totlen -= optlen;
 	}
 
+	/* All options must be read/processed and option count should be zero.
+	 * If the option count is non-zero, which means there is unread options
```

Comment:
Your comment (and commit message btw too) talks about "option count" (apparently counted in options), but actually deals with option length (apparently counted in bytes). It's not good when comments don't match the code.